cluster/v1/types.go

```diff
@@ -170,3 +170,52 @@ type ManagedClusterList struct {
 	// Items is a list of managed cluster.
 	Items []ManagedCluster `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope="Cluster"
+
+// ManagedClusterSet is a set of managed clusters.
+type ManagedClusterSet struct {
```

Comment:
I recommend providing a status with conditions so you could report things like whether the label selector was invalid and perhaps the number of clusters selected.

Reply:
Added.